docs/worldgen/biomemodifier.md

```diff
@@ -0,0 +1,648 @@
+    );
+
+// For some RegistrySetBuilder BUILDER
+//   being passed to DatapackBuiltinEntriesProvider
+//   in a listener for GatherDataEvent
+BUILDER.add(NeoForgeRegistries.Keys.BIOME_MODIFIERS, bootstrap -> {
+    // Lookup any necessary registries
+    // Static registries only need to be looked up if you need to grab the tag data
+    HolderGetter<Biome> biomes = bootstrap.lookup(Registries.BIOME);
+    HolderGetter<EntityType<?>> entities = bootstrap.lookup(Registries.ENTITY_TYPE);
+
+    // Register the Biome Modifiers
+
+    bootstrap.register(ADD_SPAWN_COSTS_EXAMPLE,
+        new AddSpawnCostsBiomeModifier(
+            // The biome(s) to add the spawn costs to
+            biomes.getOrThrow(Tags.Biomes.IS_OVERWORLD),
+            // The entities to add the spawn costs for
+            entities.getOrThrow(EntityTypeTags.SKELETONS),
+            new MobSpawnSettings.MobSpawnCost(
+                1.0, // The energy budget
+                0.1  // The amount of charge each entity takes up from the budget
+            )
+        )
+    );
+})
+```
+
+### `RemoveSpawnCostsBiomeModifier`
+
+`RemoveSpawnsBiomeModifier` removes spawn costs for mobs from biomes. The modifier takes in the `Biome` id or tag of the biomes the spawn costs are removed from and the `EntityType` id or tag of the mobs to remove the spawn cost for.
+
+```java
+// Define keys for datapack registry objects
+
+public static final ResourceKey<BiomeModifier> REMOVE_SPAWN_COSTS_EXAMPLE =
+    ResourceKey.create(
+        NeoForgeRegistries.Keys.BIOME_MODIFIERS, // The registry this key is for
+        ResourceLocation.fromNamespaceAndPath(MOD_ID, "remove_spawn_costs_example") // The registry name
+    );
+
+// For some RegistrySetBuilder BUILDER
+//   being passed to DatapackBuiltinEntriesProvider
+//   in a listener for GatherDataEvent
+BUILDER.add(NeoForgeRegistries.Keys.BIOME_MODIFIERS, bootstrap -> {
+    // Lookup any necessary registries
+    // Static registries only need to be looked up if you need to grab the tag data
+    HolderGetter<Biome> biomes = bootstrap.lookup(Registries.BIOME);
+    HolderGetter<EntityType<?>> entities = bootstrap.lookup(Registries.ENTITY_TYPE);
+
+    // Register the Biome Modifiers
+
+    bootstrap.register(REMOVE_SPAWN_COSTS_EXAMPLE,
+        new RemoveSpawnCostsBiomeModifier(
+            // The biome(s) to remove the spawnc costs from
+            biomes.getOrThrow(Tags.Biomes.IS_OVERWORLD),
+            // The entities to remove spawn costs for
+            entities.getOrThrow(EntityTypeTags.SKELETONS)
+        )
+    );
+})
+```
+
+## Creating Custom Biome Modifiers:
+
+Biome Modifiers are made up of three parts:
+
+- The [datapack registered][datareg] `BiomeModifier` used to modify the biome builder.
+- The [statically registered][staticreg] `MapCodec` that encodes and decodes the modifiers.
+- The JSON that constructs the `BiomeModifier`, using the registered id of the `MapCodec` as the indexable type.
```

Comment:
Ok, this is down here but is referenced above. It reads weirdly because of it.

Reply:
Does the datagen section need this? I thoguht this was specific to only custom biome modifiers and not needed to be referenced in datagen. But I can move it up

Reply:
So the issue when I read the above section was that you reference the example biome modifier in the datagen example, which wasn't defined yet. That's why I was confused.